Tests cleanup

fbourigault opened the pull request:
This aim to clean the test suite to reduce tests complexity.

@stof is it what you have in mind for #96?

- [x] Rewrite MailGun tests.
- [x] Rewrite Mandrill tests.
- [x] Rewrite Postmark tests.
- [x] Rewrite SendGrid tests.

stof commented on tests/Stampie/Tests/Mailer/MailGunTest.php:
```diff
@@ -2,120 +2,100 @@
+    /**
+     * @var HttpClient|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $httpClient;
```
why not using the `$adapter` property you added in the parent ?

fbourigault replied:
The idea is to remove the adapter property from the parent and keep only helper methods in the parent class. Having an HttpClient is an implementation detail.

fbourigault commented:
In https://github.com/Stampie/Stampie/pull/97/commits/cf896b2c69f213f10103e132daf82f95d4a295bd I removed the `BaseMailerTest` and moved all mock related helper methods to a new `TestCase` class which can be used by mailer and non-mailer tests.

fbourigault commented:
Mailer test rewrite is now done. If this fine, I will be able to go forward and kill the Mailer abstract class.

fbourigault commented:
@Stampie/collaborators does it looks good to you?

Nyholm commented:
Awesome
